GetStream/stream-chat-android
# [CAS-684] Hide MutableLiveDatas in View Models

https://stream-io.atlassian.net/browse/CAS-684

### 🎯 Goal
Hide `MutableLiveData` in view models

### 🛠 Implementation details

1. `MessageInputViewModel::editMessage` is replaced with `MessageInputViewModel::messageToEdit`  and MessageInputViewModel::postMessageToEdit`
2. `MessageInputViewModel::repliedMessage` type is changed to `LiveData` 
3. `MessageListViewModel::mode` type is changed to `LiveData` 
4. I've added a task to update documentation after release

### 🧪 Testing

1. Check if you can edit the message
2. Check if you can reply to the message (after sending the message, it's not rendered properly - I've added a task for that)

### ☑️ Checklist


### 🎉 GIF

![giphy](https://user-images.githubusercontent.com/17440581/117136623-eac7b680-ada8-11eb-9d55-6fd642f53848.gif)
